tests/geometry/vector.test.ts

```diff
@@ -0,0 +1,107 @@
+  });
+  
+});
+
+describe("IntVector2d", () => {
+  test("Creating vector", () => {
+    expect(new IntVector2d(1 as int, 3 as int)).toStrictEqual(new IntVector2d(1 as int, 3 as int));
+    expect(new IntVector2d(2 as int, 3 as int)).not.toStrictEqual((new IntVector2d(2 as int, 2.9 as int)));
+  });
+
+  test("Equality", () => {
+    expect(new IntVector2d(1 as int, 1 as int)).toStrictEqual(new IntVector2d(1 as int, 1 as int));
+    expect(new IntVector2d(1 as int, 10 as int)).not.toStrictEqual(new IntVector2d(1 as int, 1 as int));
+    expect(new IntVector2d(1 as int, 1 as int)).not.toStrictEqual(new BoardVector2d(1 as int, 1 as int));
+  });
+
+  test("Multiplying/dividing 2 vectors", () => {
+    expect(new IntVector2d(1 as int, 3 as int).mul(new IntVector2d(2 as int, 2 as int))).toStrictEqual(new IntVector2d(2 as int, 6 as int));
+    expect(new IntVector2d(1 as int, 2 as int).div(new IntVector2d(2 as int, 2 as int))).toStrictEqual(new IntVector2d(0 as int, 1 as int));
+    expect(new IntVector2d(7 as int, 2 as int).div(new IntVector2d(3 as int, 2 as int))).toStrictEqual(new IntVector2d(2 as int, 1 as int));
+  });
+
+  test("Multiplying/dividing vector and scalar", () => {
+    expect(new IntVector2d(1 as int, 1 as int).mul(3.5)).toStrictEqual(new IntVector2d(3 as int, 3 as int));
+    expect(new IntVector2d(5 as int, 2 as int).div(2)).toStrictEqual(new IntVector2d(2 as int, 1 as int));
+  });
+
+  test("Copy vector", () => {
+    expect(new IntVector2d(1 as int, 1 as int).copy()).toStrictEqual(new IntVector2d(1 as int, 1 as int));
+    expect(new IntVector2d(1.5 as int, 2 as int).copy()).toStrictEqual(new IntVector2d(1 as int, 2 as int));
+  });
+});
+
+describe("BoardVector2d", () => {
+  test("Creating BoardVector2d from other type vectors", () => {
+    expect(BoardVector2d.fromVector2d(new Vector2d(1.5 as int, 3 as int))).toStrictEqual(new BoardVector2d(1 as int, 3 as int));
+    expect(BoardVector2d.fromVector2d(new IntVector2d(4 as int, 1 as int))).toStrictEqual(new BoardVector2d(4 as int, 1 as int));
+  });
+
+  test("Equality", () => {
+    expect(new BoardVector2d(1 as int, 1 as int)).toStrictEqual(new BoardVector2d(1 as int, 1 as int));
+    expect(new BoardVector2d(1 as int, 10 as int)).not.toStrictEqual(new BoardVector2d(1 as int, 1 as int));
+    expect(new BoardVector2d(1 as int, 1 as int)).not.toStrictEqual(new IntVector2d(1 as int, 1 as int));
+  });
+
+
+  test("ToString functions", () => {
+    expect(new BoardVector2d(1 as int, 3 as int).xToString()).toStrictEqual("b");
+    expect(new BoardVector2d(1 as int, 3 as int).yToString()).toStrictEqual("4");
+    expect(new BoardVector2d(1 as int, 3 as int).toString()).toStrictEqual("b4");
+  });
+
+
+  test("Copy vector", () => {
+    expect(new BoardVector2d(1 as int, 1 as int).copy()).toStrictEqual(new BoardVector2d(1 as int, 1 as int));
+    expect(new BoardVector2d(1 as int, 2 as int).copy()).toStrictEqual(new BoardVector2d(1 as int, 2 as int));
+  });
+});
```

Comment:
Should test also edge cases, when these vectors aren't constructed by integer numbers.